### ISSUE TITLE
Sample (./mad deps compile plan repl) fails

### ISSUE THREAD

**Issue by ZelphirKaltstahl** [synrc/n2o#297]
On Xubuntu 16.04.3 LTS (64bit):

    ==> dependency: "git://github.com/uwiger/gproc" tag: {tag,"0.6.1"}
    ==> dependency: "git://github.com/synrc/rest" tag: {tag,"2.9"}
    ==> dependency: "git://github.com/evanmiller/erlydtl" tag: {tag,"0.8.0"}
    ==> dependency: "git://github.com/synrc/nitro" tag: {tag,"2.4"}
    ==> dependency: "git://github.com/synrc/mad" tag: {tag,"1.9"}
    ==> dependency: "git://github.com/synrc/sh" tag: {tag,"1.9"}
    ==> dependency: "git://github.com/synrc/fs" tag: {tag,"1.9"}
    ==> dependency: "git://github.com/synrc/active" tag: {tag,"1.9"}
    ==> dependency: "git://github.com/synrc/avz" tag: {tag,"4.4"}
    ==> dependency: "git://github.com/synrc/n2o" tag: {tag,"4.4"}
    ==> dependency: "git://github.com/extend/cowboy" tag: {tag,"1.0.1"}
    ==> dependency: "git://github.com/ninenines/cowlib.git" tag: "1.0.0"
    ==> dependency: "git://github.com/ninenines/ranch.git" tag: "1.0.0"
    ==> dependency: "git://github.com/synrc/kvs" tag: {tag,"3.9.1"}
    ==> dependency: "git://github.com/tim/erlang-oauth" tag: {tag,"v1.6.0"}
    ==> "gproc"
    ==> "rest"
    ==> "erlydtl"
    escript: exception error: undefined function yecc:file/1
      in function  mad_yecc:compile/5 (/Users/5HT/depot/synrc/mad/src/compile/mad_yecc.erl, line 11)
      in call from mad_compile:compile_files/5 (/Users/5HT/depot/synrc/mad/src/compile/mad_compile.erl, line 79)
      in call from mad_compile:dep/4 (/Users/5HT/depot/synrc/mad/src/compile/mad_compile.erl, line 68)
      in call from mad_compile:deps/4 (/Users/5HT/depot/synrc/mad/src/compile/mad_compile.erl, line 20)
      in call from mad_compile:compile/1 (/Users/5HT/depot/synrc/mad/src/compile/mad_compile.erl, line 8)
      in call from mad:'-main/1-fun-3-'/3 (/Users/5HT/depot/synrc/mad/src/mad.erl, line 19)
      in call from lists:foldl/3 (lists.erl, line 1262)

`inotify-tools` are already installed.

I could not find `yecc` with `sudo aptitude search erlang-`, so I guess it's something n2o specific.

**Comment by 5HT** [comment]
yecc — is a default Erlang parser generator. Any Erlang installation without it — wasted!

**Comment by ZelphirKaltstahl** [comment]
@5HT Does that mean I have to compile Erlang? (I already looked for a guide, but have not started yet)

And what does:

> And mad is only compatible with synrc stack, not tim/erlang-oauth

mean? I don't know what the synrc stack is, nor what tim/erlang-oauth is. I am only learning Erlang at the moment and would like to try n2o soon, when I am a bit comfortable with using Erlang. I don't necessarily know much about packaging in Erlang or dependencies of some libraries. Please bear with a beginner : )

**Comment by ZelphirKaltstahl** [comment]
The issue went away when using self compiled Erlang. A lot of warnings appeared, but the example ultimately did run.

**Comment by 5HT** [comment]
where did you get `tim/erlang-oauth` — this is not shipped with n2o neither its samples

**Comment by ZelphirKaltstahl** [comment]
> where did you get tim/erlang-oauth

> I don't know what the synrc stack is, nor what tim/erlang-oauth is.

I also don't know where it comes from ; ) I guess from the Ubuntu distribution of Erlang.

> on linux we usually install erlang with https://github.com/kerl/kerl

Oh good to know. I now already built it from source with the procedure described in the readme with `make` and such. Is that a bad idea, compared to building it with `kerl`?

> we draft our synrc libs to be compatible with mad our tiny and fast package manager

Aha, I thought `mad` was merely a compilation tool.

I'll have to get into the whole tooling, so this is useful information. I did not find `kerl`, nor did I find `mad`, when looking for how to build Erlang or Erlang package managers respectively.
Is there a page somewhere, which lists many of these tools?
